File: 2006/J062.pas
var
  valid: array[1..100, 1..100] of boolean;
  n, m, k, i, j, from, too, pieces, a: longint;
  ok:boolean;
begin
  read(n, m, k);
  for i := 1 to m do
    for j := 1 to n do
      valid[i][j] := i = 1;
  for i := 1 to k do
  begin
    read(from, too, pieces);
    for j := 1 to pieces do
    begin
      read(a);
      valid[too][a] := valid[too][a] or valid[from][a]
    end
  end;
  for i := 1 to m do
  begin
    ok := true;
    for j := 1 to n do
      ok := ok and valid[i][j];
    if (ok) then
      writeln(i)
  end
end.

File: 2006/J062.cpp
#include <cstdio>
bool valid[101][101];
int main() {
  int n, m, k;
  scanf("%d %d %d", &n, &m, &k);
  for (int i = 1; i <= m; i++)
    for (int j = 1; j <= n; j++)
      valid[i][j] = (i == 1);
  for (int i = 0; i < k; i++) {
    int from, to, pieces, a;
    scanf("%d %d %d", &from, &to, &pieces);
    for (int j = 0; j < pieces; j++) {
      scanf("%d", &a);
      valid[to][a] |= valid[from][a];
    }
  }
  for (int i = 1; i <= m; i++) {
    bool ok = true;
    for (int j = 1; j <= n; j++)
      ok &= valid[i][j];
    if (ok)
      printf("%d\n", i);
  }
  return 0;
}
